web/src/components/vis/VolcanoPlot.vue

```diff
@@ -23,106 +46,146 @@ export default {
-        .attr('class', 'x-threshold')
-        .style('stroke', 'black')
-        .style('stroke-width', 0.5)
-        .attr('x1', (d) => this.scaleX(d * this.minFoldChange))
-        .attr('y1', 0)
-        .attr('x2', (d) => this.scaleX(d * this.minFoldChange))
-        .attr('y2', this.height - this.margin.top - this.margin.bottom);
-      svg.select('.plot').selectAll('line.y-threshold')
-        .data([1])
-        .join('line')
-        .attr('class', 'y-threshold')
-        .style('stroke', 'black')
-        .style('stroke-width', 0.5)
-        .attr('x1', 0)
-        .attr('y1', () => this.scaleY(this.minLogP))
-        .attr('x2', this.width - this.margin.left - this.margin.right)
-        .attr('y2', () => this.scaleY(this.minLogP));
-    },
+    }) as Ref<[number, number]>;
+
+    const scaleX = computed(() => scaleLinear()
+      .domain(xrange.value)
+      .range([0, dwidth.value]));
+
+    const scaleY = computed(() => scaleLinear()
+      .domain(yrange.value)
+      .range([dheight.value, 0]));
+
+    const axisX = computed(() => axisBottom(scaleX.value));
+    const axisY = computed(() => axisLeft(scaleY.value));
+
+    onMounted(() => {
+      watchEffect(() => {
+        // Recalculate width/height
+        const bb = mainRef.value.getBoundingClientRect();
+        width.value = bb.width;
+
+        const svg = select(svgRef.value);
+        axisPlot(svg, axisX.value, axisY.value);
+
+        const _scaleX = scaleX.value;
+        const _scaleY = scaleY.value;
+
+        // @ts-ignore
+        svg.select('.plot').selectAll('circle')
+          .data<TransformedRow>(transformedRows.value)
+          .join((enter) => {
+            const r = enter.append('circle');
+            r.append('title');
+            return r;
+          })
+          .attr('r', (d) => (Math.abs(d.x) >= props.minFoldChange && d.y >= props.minLogP ? radius * 2 : radius))
+          .attr('opacity', (d) => (Math.abs(d.x) >= props.minFoldChange && d.y >= props.minLogP ? 1 : 0.5))
+          .attr('cx', (d) => _scaleX(d.x))
+          .attr('cy', (d) => _scaleY(d.y))
+          .style('fill', (d) => d.color)
+          .select('title')
+          .text((d) => `${d.name}: ${d.log2FoldChange} x ${d.pValue}`);
+        svg.select('.plot').selectAll('line.x-threshold')
+          .data([-1, 1])
+          .join('line')
+          .attr('class', 'x-threshold')
+          .style('stroke', 'black')
+          .style('stroke-width', 0.5)
+          .attr('x1', (d) => _scaleX(d * props.minFoldChange))
+          .attr('y1', 0)
+          .attr('x2', (d) => _scaleX(d * props.minFoldChange))
+          .attr('y2', height.value - margin.top - margin.bottom);
+
+        svg.select('.plot').selectAll('line.y-threshold')
+          .data([1])
+          .join('line')
+          .attr('class', 'y-threshold')
+          .style('stroke', 'black')
+          .style('stroke-width', 0.5)
+          .attr('x1', 0)
+          .attr('y1', () => _scaleY(props.minLogP))
```

Comment:
This is a function computing a static value; it seems like you could just remove the function call altogether:

```suggestion
          .attr('y1', _scaleY(props.minLogP))
```